src/metabase/driver/sql_jdbc/sync/describe_database.clj

```diff
@@ -89,32 +92,45 @@
 
   This is as much as 15x faster for Databases with lots of system tables than `post-filtered-active-tables` (4 seconds
   vs 60)."
-  [driver ^Connection conn & [db-name-or-nil]]
+  [driver ^Connection conn & [db-name-or-nil schema-inclusion-filters schema-exclusion-filters]]
   {:pre [(instance? Connection conn)]}
   (let [metadata (.getMetaData conn)]
     (eduction
      (comp (mapcat (fn [schema]
                      (db-tables driver metadata schema db-name-or-nil)))
            (filter (fn [{table-schema :schema, table-name :name}]
-                     (i/have-select-privilege? driver conn table-schema table-name))))
-     (i/syncable-schemas driver conn metadata))))
+                     (and
+                       (driver.s/include-schema? schema-inclusion-filters schema-exclusion-filters table-schema)
+                       (i/have-select-privilege? driver conn table-schema table-name)))))
+     (i/syncable-schemas driver conn metadata schema-inclusion-filters schema-exclusion-filters))))
```

Comment:
This is a bit confusing to me. We filter `(and (driver.s/include-schema? inclusion-filters))` but also, the original seq is `(i/syncable-schemas inclusion-filters)`. It seems we are filtering twice. 

`i/syncable-schemas` for `:sql-jdbc`
```clojure
(defmethod i/syncable-schemas :sql-jdbc
  [driver _ metadata schema-inclusion-patterns schema-exclusion-patterns]
  (eduction (remove (set (i/excluded-schemas driver)))
            (filter (partial driver.s/include-schema? schema-inclusion-patterns schema-exclusion-patterns))
            (all-schemas metadata)))
```
which removes and includes.

For redshift, 
```clojure
(defmethod sql-jdbc.sync/syncable-schemas :redshift
  [driver conn metadata schema-inclusion-patterns schema-exclusion-patterns]
  (reducible-schemas-with-usage-permissions
   conn
   (eduction
    (remove (set (sql-jdbc.sync/excluded-schemas driver)))
    (filter (partial driver.s/include-schema? schema-inclusion-patterns schema-exclusion-patterns))
    (sync.describe-database/all-schemas metadata))))
```

So it seems that `i/syncable-schemas` runs the schemas through redshifts select privileges and then again through `:sql-jdbc`'s `i/have-select-privilege? :sql-jdbc` again, as well as the schema filtering. It seems there's a lot going on here, and lots of it twice.

This code is also hard to follow because of potemkin. Redshift implements `sql-jdbc.sync/syncable-schemas :redshift` but otherwise the default is `i/have-select-privilege? :sql-jdbc` because `metabase.driver.sql-jdbc.sync.interface` contains the actual defmethod whereas `metabase.driver.sql-jdbc.sync` potemkin import-vars this defmethod.

Reply:
Yeah, there was definitely redundant filtering happening in the `:sql-jdbc` impl.  So I removed that.

Also, the `:redshift` impl actually did need the call to `include-schema?` since it was operating on `all-schemas`.  I changed it up to just delegate to the `:sql-jdbc` impl instead, and only apply the `reducible-schemas-with-usage-permissions` on top of those.

Reply:
> This code is also hard to follow because of Potemkin.

`M-.` still jumps to the actual definition regardless of whether there's a Potemkin alias for it or not